packages/payment-processor/src/payment/any-to-erc20-proxy.ts

```diff
@@ -23,7 +23,7 @@ import { IPreparedTransaction } from './prepared-transaction';
  * - maxToSpend: maximum number of tokens to be spent when the conversion is made
  */
 export interface IPaymentSettings {
-  currency: RequestLogicTypes.ICurrency;
+  currency?: RequestLogicTypes.ICurrency;
```

Comment:
I don't understand, what's the point of making it optional and then throwing if it's undefined ?